### PR TITLE
PP-6960 Support setting MOTO mask properties on gateway accounts

### PR DESCRIPTION
**Pull request by alexbishop1**
Support sending PATCH requests to `/v1/api/accounts/{accountId}` with JSON bodies like this:

```json
{
  "op": "replace",
  "path": "moto_mask_card_number_input",
  "value": true
}
```

… or this:

```json
{
  "op": "replace",
  "path": "moto_mask_card_security_code_input",
  "value": false
}
```

### CODE REVIEW

File: src/main/java/uk/gov/pay/connector/gatewayaccount/service/GatewayAccountService.java [hunk 1]
```diff
@@ -88,50 +91,83 @@ public GatewayAccountResponse createGatewayAccount(GatewayAccountRequest gateway
...
+            ),
+            entry(
+                FIELD_MOTO_MASK_CARD_NUMBER_INPUT,
+                (gatewayAccountRequest, gatewayAccountEntity) -> gatewayAccountEntity.setMotoMaskCardNumberInput(gatewayAccountRequest.valueAsBoolean())
+            ),
+            entry(
+                FIELD_MOTO_MASK_CARD_SECURITY_CODE_INPUT,
+                (gatewayAccountRequest, gatewayAccountEntity) -> gatewayAccountEntity.setMotoMaskCardSecurityCodeInput(gatewayAccountRequest.valueAsBoolean())
+            )
+    );
```
Comment (oswaldquek) [inline src/main/java/uk/gov/pay/connector/gatewayaccount/service/GatewayAccountService.java:170]: nice refactor

File: src/test/java/uk/gov/pay/connector/service/GatewayAccountServiceTest.java [hunk 2]
```diff
@@ -274,7 +287,63 @@ public void shouldUpdateAllowMotoFalse() {
...
+        Optional<GatewayAccount> optionalGatewayAccount = gatewayAccountService.doPatch(GATEWAY_ACCOUNT_ID, request);
+        assertThat(optionalGatewayAccount.isPresent(), is(true));
+        verify(mockGatewayAccountEntity).setMotoMaskCardNumberInput(false);
+        verify(mockGatewayAccountDao).merge(mockGatewayAccountEntity);
+    }
+
+    @Test
+    public void shouldUpdateMotoMaskCardSecurityCodeInputTrue() {
+        JsonPatchRequest request = JsonPatchRequest.from(new ObjectMapper().valueToTree(Map.of(
+                "op", "replace",
```
Comment (oswaldquek) [inline src/test/java/uk/gov/pay/connector/service/GatewayAccountServiceTest.java:322]: Doesn't stop the approval of this PR but I was wondering if we should test the operations of "add" and "remove" too. I realise we haven't tested these operations for the other things though.

**Review by oswaldquek — APPROVED** [review]
Optional comment to be addressed

File: src/test/java/uk/gov/pay/connector/service/GatewayAccountServiceTest.java (hunk 2 above)
Comment (alexbishop1) [reply src/test/java/uk/gov/pay/connector/service/GatewayAccountServiceTest.java:322]: We don’t actually support them (we cheat and just do everything with `"replace"`) so not much point. (Ideally, we’d try to make our JSON Patch implementation more conformant at some stage!)

**Pull request merged**